pagopa/pn-frontend
# Feat(PN/6792):  analog progress event PNALL001

## Short description
Show the analog-flow timeline events having `deliveryDetailCode = PNALL001`, specify the copy for the single- and multi-recipient cases.

## List of changes proposed in this pull request
- Added `PNALL001` to the list of `deliveryDetailCode` values for which analog flow events are shown.
- Added copy for these events in `notifiche.json` for PA / PF / PG.
- Adapted the way address are managed to show in timeline events, since analog-flow events must obtain the address info in a way which differs from that used in the other events that include address in the copy (namely SEND_ANALOG_DOMICILE and SEND_SIMPLE_REGISTERED_LETTER).

## How to test
Access the detail of a notification including a PNALL001 analog-flow event, check that such event is indeed shown, with the copy specified in https://pagopa.atlassian.net/wiki/spaces/PN/pages/629014723/Timeline+events+e+cosa+mostrare+agli+utenti .  

Unfortunately, I know of no notifications including such events at the time of writing this description (2023.06.30 12:20). Hence I had to mock the notification API to test what I did.  
This is the code I added to `getSentNotification` (for PA) / `getReceivedNotification` (for PF)
```
      if (response.data.iun === 'HGZK-XLJQ-ZTGL-202306-Z-1') {
        const elementToModify = response.data.timeline.find(elem => elem.elementId === 'SEND_ANALOG_PROGRESS.IUN_HGZK-XLJQ-ZTGL-202306-Z-1.RECINDEX_0.ATTEMPT_1.IDX_1');
        if (elementToModify) {
          // eslint-disable-next-line functional/immutable-data
          (elementToModify.details as any).deliveryDetailCode = 'PNALL001';
        }
      } else if (response.data.iun === 'YTJQ-TWTW-NLWH-202306-K-1') {
        const elementToModify = response.data.timeline.find(elem => elem.elementId === 'SEND_ANALOG_PROGRESS.IUN_YTJQ-TWTW-NLWH-202306-K-1.RECINDEX_0.ATTEMPT_1.IDX_1');
        if (elementToModify) {
          // eslint-disable-next-line functional/immutable-data
          (elementToModify.details as any).deliveryDetailCode = 'PNALL001';
        }
      }
```
The involved notifications, present in TEST environment, are:
- HGZK-XLJQ-ZTGL-202306-Z-1 - Palermo / leonardo
- YTJQ-TWTW-NLWH-202306-K-1 - sappada / leonardo + ada - used to check multi-recipient copy (accessing as PA).

Of course the mocks are not included in this PR.

It would be nice to also verify that the events of category SEND_ANALOG_DOMICILE and SEND_SIMPLE_REGISTERED_LETTER are still being shown appropriately.  
For SEND_ANALOG_DOMICILE you can check with any analog-flow notification, including the two earlier examples.
For SEND_SIMPLE_REGISTERED_LETTER, in TEST we have
- ZVKU-VAQH-NDVT-202306-L-1 Palermo (even if the sender description refers to Milano) / cristoforocolombo
